infrastructure/src/main/java/org/corfudb/infrastructure/DataStore.java

```diff
@@ -129,7 +147,8 @@ public synchronized void write(@Nonnull String key, @Nonnull Object value) {
                                     StandardOpenOption.TRUNCATE_EXISTING, StandardOpenOption.SYNC);
                             Files.move(tmpPath, path, StandardCopyOption.REPLACE_EXISTING,
                                     StandardCopyOption.ATOMIC_MOVE);
-                            syncDirectory(logDir);
+                            syncDirectory(logDirPath);
+                            prefixesToClean.forEach(prefix -> dataStoreFileCleanup(prefix, key));
```

Comment:
Add a comment on why its "fine" to invoke this clean up (process) on every put.

Reply:
Done.